akka-projection-core/src/main/scala/akka/projection/internal/InternalProjectionState.scala

```diff
@@ -370,20 +371,28 @@ private[projection] abstract class InternalProjectionState[Offset, Envelope](
   }
 
   def mappedSource(): Source[Done, Future[Done]] = {
-
     val handlerLifecycle = handlerStrategy.lifecycle
     statusObserver.started(projectionId)
     telemetry = TelemetryProvider.start(projectionId, system)
 
     val source: Source[ProjectionContextImpl[Offset, Envelope], NotUsed] =
       Source
-        .futureSource(
-          handlerLifecycle
-            .tryStart()
-            .flatMap { _ =>
-              sourceProvider
-                .source(() => readOffsets())
-            })
+        .futureSource(readPaused().flatMap {
+          case false =>
+            logger.debug("Projection [{}] started in resumed mode.", projectionId)
+            handlerLifecycle
+              .tryStart()
+              .flatMap { _ =>
+                sourceProvider
+                  .source(() => readOffsets())
+              }
+          case true =>
+            logger.info("Projection [{}] started in paused mode.", projectionId)
+            // paused stream, no elements
+            statusObserver.paused(projectionId)
+            telemetry.paused()
+            Future.successful(Source.never[Envelope])
```

Comment:
This trick of swapping to a `Source.never` is neat. 
When restarting a node we don't need management code to take decisions, we can just start the actor and feed it with a this `Source.never`. 

That's clever!